src/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -349,6 +357,10 @@ public void mergeAlignment(final File referenceFasta) {
                 final boolean clone = nextAligned.numHits() > 1 || nextAligned.hasSupplementalHits();
                 SAMRecord r1Primary = null, r2Primary = null;
 
+                // by this point there should be a single chosen primary alignment, which we will use to determine whether the read is contaminant.
+                // this must be done before the main iteration, since secondary / supplementary alignments will be affected by the primary.
+                final boolean isContaminant = this.unmapContaminantReads && isContaminant(nextAligned);
```

Comment:
can you rename this variable to that it includes the unmapContaminantReads somehow?

perhaps removeDueToContaminant, or removeSinceContaminant